src/components/TopNav/NavMenu.js

```diff
@@ -8,11 +8,15 @@ import NavigationMenu from 'material-ui/svg-icons/navigation/menu'
 import IconButton from 'material-ui/IconButton'
 import * as auth from 'services/authNService'
 
-var transformIdToTicketLink = (id, index) =>
-  <MenuItem
-    key={'ticket-' + id + '-index-' + index}
-    primaryText={<Link to={'/tickets/' + id} >{'Ticket ' + id}</Link>}
-  />
+var transformIdToTicketLink = (id, index) => {
+  let addCurrentPageToHistory = () => { history.push(this.to) /* global history */ }
```

Comment:
[This comment](https://github.com/Azure/Sia-EventUI/pull/75/commits/2e2b95fe457b341b781319c28cfb07bbee5bf945#r165778005) indicates we should inject this change at the Link level. Initial investigation didn't yield an obvious way to achieve this.

Reply:
I think that the Timeline/Event components failing to load appropriate content [has something to do with blocked updates.](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/blocked-updates.md)

I attempted to brute force this by wrapping most `connect()` calls in `withRouter`, but this does not resolve the issue. I think I need to pair with someone hwo has greatr React knowledge to correctly resolve this.